tokenizers/src/utils/mod.rs

```diff
@@ -5,6 +5,7 @@ pub mod iter;
 pub mod padding;
 pub mod parallelism;
 pub(crate) mod progress;
+pub mod trie;
```

Comment:
I'm not very fond of the `utils` module, it contains a lot of different things that aren't really related.

Wouldn't a separate `trie` module work ?

Reply:
I don't think it's worth it. This is a pretty small helper struct. 

`utils` is definitely not the best, specially for "core"  things, but this is not IMHO. Don't think it's valuable enough to be on its own. If at all, it would be a separate crate (which it's probably too small to be the case).

Actually `aho-corasick` could be used in the end (There's `Longest` option which I missed in the doc in my initial inquiry)